File: resources/xcode/NativePHP/NativeRender/ActivityIndicatorRenderer.swift
import UIKit

struct ActivityIndicatorViewRenderer: NativeViewRenderer {
    func createView(node: NativeUINode) -> UIView {
        let style: UIActivityIndicatorView.Style
        switch node.props.getInt("size") {
        case 1: style = .large
        case 2: style = .medium
        default: style = .medium
        }

        let indicator = UIActivityIndicatorView(style: style)
        indicator.hidesWhenStopped = false
        indicator.startAnimating()

        let color = node.props.getColor("color", default: 0)
        if color != 0 {
            indicator.color = UIColor(argb: color)
        }

        return indicator
    }

    func updateView(_ view: UIView, node: NativeUINode) {
        guard let indicator = view as? UIActivityIndicatorView else { return }
        let color = node.props.getColor("color", default: 0)
        if color != 0 {
            indicator.color = UIColor(argb: color)
        }
        if !indicator.isAnimating {
            indicator.startAnimating()
        }
    }
}

File: resources/androidstudio/app/src/main/java/com/nativephp/mobile/ui/nativerender/ActivityIndicatorRenderer.kt
package com.nativephp.mobile.ui.nativerender

import android.content.Context
import android.content.res.ColorStateList
import android.view.View
import android.widget.ProgressBar

class ActivityIndicatorViewRenderer : NativeViewRenderer {
    override fun createView(context: Context, node: NativeUINode): View {
        val style = resolveStyle(node)
        val pb = ProgressBar(context, null, style)
        pb.isIndeterminate = true
        applyProps(pb, node)
        return pb
    }

    override fun updateView(view: View, node: NativeUINode) {
        val pb = view as? ProgressBar ?: return
        applyProps(pb, node)
    }

    private fun applyProps(pb: ProgressBar, node: NativeUINode) {
        val color = node.props.getColor("color", 0)
        if (color != 0) {
            pb.indeterminateTintList = ColorStateList.valueOf(color)
        }
    }

    private fun resolveStyle(node: NativeUINode): Int {
        return when (node.props.getInt("size")) {
            1 -> android.R.attr.progressBarStyleLarge
            2 -> android.R.attr.progressBarStyleSmall
            else -> android.R.attr.progressBarStyle
        }
    }
}
